client/components/overviewWidget/StyleSelector.jsx

```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+import Styles from './Styles.jsx';
+import axios from 'axios';
+
+class StyleSelector extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      product: null,
+      styles: [],
+      selectedStyle: ''
+    };
+  }
+
+  componentDidMount() {}
+
+  componentDidUpdate(prevProps) {
+    if (this.props.product !== prevProps.product) {
+      axios.get(`/api/fec2/hrnyc/products/${this.props.product.id}/styles`)
+        .then(({data}) => {
+          this.setState({
+            product: this.props.product,
```

Comment:
Making sure you mean to be setting state from your props as that practice is discouraged by React.